docs/token-builder.md

```diff
@@ -0,0 +1,288 @@
+
+Each of the token classes has its own set of validations which are run when the token is instantiated. Any errors are added to an `errors` property on that instance of the token class.
+
+In `sorcerer`, the Token Parser will extract the tokens from request body and replace any occurrence of the token in an API data source's `url`, `body`, or `headers` with the token's computed value.
+
+## Token Types
+
+The Token Builder API includes several token utility classes, each serving a unique purpose.
+
+Currently supported tokens are:
+
+- ReplaceToken
+- ReplaceLargeToken
+- SecretToken
+- HmacToken
+- Sha1Token
+
+### ReplaceToken
+
+Replaces token with the `value` included in the token options. The length of this value must be less than 100 characters.
+
+Example:
+
+```jsx
+const options = {
+  name: 'FavoriteBand',
+  cacheOverride: 'Movable Band',
+  value: 'Beatles',
+  skipCache: true,
+};
+
+const tokenModel = new ReplaceToken(options);
+
+tokenModel.toJSON() // returns:
+// {
+//   name: 'FavoriteBand',
+//   type: 'replace',
+//   cacheOverride: 'Movable Band',
+//   skipCache: true,
+//   value: 'Beatles',
+// };
+```
+
+### ReplaceLargeToken
+
+Functionally, this behaves the same as the `ReplaceToken`, with the difference being that this token should only be used if the length of the replace value is greater than a specified character limit
+
+**Example:**
+
+```jsx
+const replaceValue = "some really long string"
+
+const options = {
+  name: 'FavoriteBand',
+  cacheOverride: 'Movable Band',
+  value: replaceValue,
+  skipCache: true,
+};
+
+const tokenModel = new ReplaceLargeToken(options);
+
+tokenModel.toJSON() // returns:
+// {
+//   name: 'FavoriteBand',
+//   type: 'replace',
+//   cacheOverride: 'Movable Band',
+//   skipCache: true,
+//   value: 'some really long string',
+// };
+```
+
+### SecretToken
+
+Replaces token with the decrypted value of the a secret, where `path` is the name of a secret that was created for a data source in the data source wizard.
+
+**Example:**
+
+```jsx
+const options = { name: 'myApiKey', path: 'watson', skipCache: true, cacheOverride: 'xyz' };
+const tokenModel = new SecretToken(options);
+
+// tokenModel.toJSON() returns:
+// {
+//   name: 'myApiKey',
+//   type: 'secret',
+//   path: 'watson',
+//   skipCache: true,
+//   cacheOverride: 'xyz',
+// };
+
+```
+
+### HmacToken
+
+Replaces HMAC token with an HMAC signature generated from the token options
+
+O**ptions:**
+
+- **stringToSign**
+
+    String that represents the request & will be used when generating HMAC signature
+
+- **algorithm**
+
+    The hashing algorithm: `sha1` , `sha256`, `md5`
+
+    (for now we only support these 3, but can easily add more later if needed)
+
+- **secretName**
+
+    Name of the data source secret (ex: `watson`)
+
+- **encoding**
+
+    Once the signature is generated it needs to be encoded
+
+    The following encodings are supported:`hex`, `base64`,`base64url` ,`base64percent`
```

Comment:
fix: I think the spacing on this sentence is a bit off between the `:` and the `,`.